bot.user.js

```diff
@@ -1325,14 +1329,20 @@ var userInterface = window.userInterface = (function() {
             } else if (bot.isBotEnabled && bot.isBotRunning) {
                 bot.isBotRunning = false;
                 if (window.lastscore && window.lastscore.childNodes[1]) {
-                    bot.scores.push(parseInt(window.lastscore.childNodes[1].innerHTML));
+                    var score = parseInt(window.lastscore.childNodes[1].innerHTML);
+                    var duration = Date.now() - bot.startTime;
+                    bot.scores.push({
+                        score: score,
+                        duration: duration
+                    });
                     bot.scores.sort(function(a, b) {
-                        return b - a;
+                        return b.score - a.score;
                     });
                     userInterface.updateStats();
                 }
 
                 if (window.autoRespawn) {
+                    bot.startTime = Date.now();
```

Comment:
This change suggests bad state management. We should have some bot states to check our code against
- Not playing
- Connecting
- Connected?
- Playing
- Dying